shell/common/shell.cc

```diff
@@ -223,77 +223,59 @@ void Shell::InitUIThread() {
 }
 
 void Shell::AddRasterizer(const ftl::WeakPtr<Rasterizer>& rasterizer) {
-  FTL_DCHECK(gpu_thread_checker_ &&
-             gpu_thread_checker_->IsCreationThreadCurrent());
+  std::lock_guard<std::recursive_mutex> lk(mutex_);
   rasterizers_.push_back(rasterizer);
 }
 
 void Shell::PurgeRasterizers() {
-  FTL_DCHECK(gpu_thread_checker_ &&
-             gpu_thread_checker_->IsCreationThreadCurrent());
+  std::lock_guard<std::recursive_mutex> lk(mutex_);
   rasterizers_.erase(
       std::remove_if(rasterizers_.begin(), rasterizers_.end(), IsInvalid),
       rasterizers_.end());
 }
 
 void Shell::GetRasterizers(std::vector<ftl::WeakPtr<Rasterizer>>* rasterizers) {
-  FTL_DCHECK(gpu_thread_checker_ &&
```

Comment:
The check for the GPU thread was replaced by a mutex, but the pointers are still ftl::WeakPtrs that are bound to the GPU thread.

Is there a need to access the rasterizers list outside the GPU thread?  If not, then I'd prefer to revert to using the GPU thread checker